inc/blocks/namespace.php

```diff
@@ -140,13 +190,13 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 				'ID' => $posts[0]->ID,
 				'post_content' => serialize_block( $xb ),
 				'post_title' => $xb['attrs']['title'] ?: $default_title,
-				'post_parent' => $post_ID,
+				'post_parent' => $post_ID ?: '',
 			] );
 		}
 
-		// Store XB type, ensure back compat with existing XBs also.
-		update_post_meta( $xb_post_id, '_xb_type', $xb['blockName'] );
-		update_post_meta( $xb_post_id, '_xb_type_' . sanitize_key( $xb['blockName'] ), true );
+		// Store XB type.
+		update_post_meta( $xb_post_id, $xb_meta_key, $xb['blockName'] );
```

Comment:
@roborourke I'm a little unsure on this comment and what I should be achieving here. I feel it's probably simple and that I am over thinking it?

Reply:
Yeah maybe overthinking a bit :) When I originally suggested adding some meta data to indicate that the XB was created as a widget I didn't mean you should change the existing post meta being saved. What was the thought process behind that change? Code reuse?

Reply:
I may not be 100% here but I think I have a better idea now, it should be added with the the `wp_insert_post` rather than with the `update_post_meta`.

I didn't think we wanted to keep both the `xb_type` post meta, but I suspect the way I had gone with then there would be too many variants. Where as now we could query the post meta by `_xb_type` or the `_xb_content`?